src/gConsent/manage/getAccessGrantAll.ts

```diff
@@ -249,15 +249,13 @@ async function getAccessGrantAll(
         (vc) => isBaseAccessGrantVerifiableCredential(vc) && isAccessGrant(vc),
       );
   }
-
-  return result.filter(
-    (vc) =>
-      // Explicitly non-recursive grants are filtered out, except if they apply
-      // directly to the target resource.
+  return result.filter((vc) => {
+    return (
```

Comment:
Its interesting that the comment you removed here seems to imply the incorrect behaviour was part of the original design approach.

Reply:
Actually I'll unerase teh coment, but updating it to make it more accurate. It's correct behavior to filter out non-recursive grants if they apply only to an ancestor of the target resource, but that only makes sense if there *is* a target resource ^^